spec/integration/data_stream/manifest.spec.yml

```diff
@@ -159,9 +159,142 @@ spec:
+                      additionalProperties: false
+                      properties:
+                        limit:
+                          description: Limit on the number of dimension fields on this data stream.
+                          type: integer
+                sort:
+                  type: object
+                  additionalProperties: false
+                  properties:
+                    field:
+                      description: Fields used to sort the document in the Lucene segments.
+                      oneOf:
+                        - type: string
+                        - type: array
+                          items:
+                            type: string
+                    order:
+                      description: The sort order to use for each field.
+                      oneOf:
+                        - type: string
+                          enum:
+                            - asc
+                            - desc
+                        - type: array
+                          items:
+                            type: string
+                            enum:
+                              - asc
+                              - desc
         mappings:
           description: Mappings section of index template
           type: object
+          additionalProperties: false
+          properties:
+            date_detection:
+              description: >
+                If true, new string fields will be checked to see if their contents match with `dynamic_date_formats`.
+                If they do, the field is dynamically added with type `date`. Set to false to disable this feature.
+              type: boolean
+            dynamic:
+              description: >
+                Control if new fields can be added dynamically. When set to `false`, new fields are not indexed.
+                When set to `strict`, fields must be explicitly added to the mapping. With `runtime`, fields are
+                not indexed, but added as runtime fields. Defaults to `true`, where Elasticsearch tries to index
+                any new field.
+              oneOf:
+                - type: boolean
+                - type: string
+                  enum:
+                    - runtime
+                    - strict
+            dynamic_date_formats:
+              # Added to support `date_detection`, but usually `date_detection` is only used to disable it.
+              description: Custom date formats to use for date detection.
+              type: array
+              items:
+                type: string
+                examples:
+                  - ["strict_date_optional_time","yyyy/MM/dd HH:mm:ss Z||yyyy/MM/dd Z"] # This is the default if not set.
+                  - ["MM/dd/yyyy"]
+
+            dynamic_templates:
+              type: array
+              items:
+                type: object
+                # maxProperties: 1 # No idea why this doesn't work. But this will be also checked by Elasticsearch.
```

Comment:
Tried to use this value, but I get this error:
> could not read root folder spec file: could not load spec for "integration/data_stream/spec.yml": could not load schema for "integration/data_stream": failed to load schema for "integration/data_stream/manifest.spec.yml": maxProperties must be of an integer

I didn't find why it could be detected as a different type than integer :/

Reply:
Workaround added in https://github.com/elastic/package-spec/pull/587/commits/2d31fd8249471108e49d3710754e1816e0a0bf92

Reply:
Thanks for adding this!